src/legacy/core_plugins/kbn_doc_views/public/views/table/table.test.tsx

```diff
@@ -0,0 +1,250 @@
+
+describe('DocViewTable at Discover', () => {
+  // At Discover's main view, all buttons are rendered
+  // check for existence of action buttons and warnings
+
+  const hit = {
+    _index: 'logstash-2014.09.09',
+    _score: 1,
+    _source: {
+      message:
+        'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. \
+        Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus \
+        et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, \
+        ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. \
+        Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, \
+        rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. \
+        Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. \
+        Phasellus ullamcorper ipsum rutrum nunc. Nunc nonummy metus. Vestibulum volutpat pretium libero. Cras id dui. Aenean ut',
+      extension: 'html',
+      not_mapped: 'yes',
+      bytes: 100,
+      objectArray: [{ foo: true }],
+      relatedContent: {
+        test: 1,
+      },
+      scripted: 123,
+      _underscore: 123,
+    },
+  };
+
+  const props = {
+    hit,
+    columns: ['extension'],
+    indexPattern,
+    filter: jest.fn(),
+    onAddColumn: jest.fn(),
+    onRemoveColumn: jest.fn(),
+  };
+  const component = mount(<DocViewTable {...props} />);
+  [
+    {
+      _property: '_index',
+      addInclusiveFilterButton: true,
+      collapseBtn: false,
+      noMappingWarning: false,
+      toggleColumnButton: true,
+      underscoreWarning: false,
+    },
+    {
+      _property: 'message',
+      addInclusiveFilterButton: false,
+      collapseBtn: true,
+      noMappingWarning: false,
+      toggleColumnButton: true,
+      underscoreWarning: false,
+    },
+    {
+      _property: '_underscore',
+      addInclusiveFilterButton: false,
+      collapseBtn: false,
+      noMappingWarning: false,
+      toggleColumnButton: true,
+      underScoreWarning: true,
+    },
+    {
+      _property: 'scripted',
+      addInclusiveFilterButton: false,
+      collapseBtn: false,
+      noMappingWarning: false,
+      toggleColumnButton: true,
+      underScoreWarning: false,
+    },
+    {
+      _property: 'not_mapped',
+      addInclusiveFilterButton: false,
+      collapseBtn: false,
+      noMappingWarning: true,
+      toggleColumnButton: true,
+      underScoreWarning: false,
+    },
+  ].forEach(check => {
+    const rowComponent = findTestSubject(component, `tableDocViewRow-${check._property}`);
+
+    it(`renders row for ${check._property}`, () => {
+      expect(rowComponent.length).toBe(1);
+    });
+
+    [
+      'addInclusiveFilterButton',
+      'collapseBtn',
+      'noMappingWarning',
+      'toggleColumnButton',
+      'underscoreWarning',
+    ].forEach(element => {
```

Comment:
You don't need to disable ts here, if specifying the array as 
```ts
([
      'addInclusiveFilterButton',
      'collapseBtn',
      'noMappingWarning',
      'toggleColumnButton',
      'underscoreWarning',
    ] as const)
```
it understands that all keys from the array exist on `check`

Reply:
very helpful, thx!